File: backend/src/main/kotlin/com/back/back9/domain/orders/controller/OrdersController.java
package com.back.back9.domain.orders.controller;

import com.back.back9.domain.orders.dto.OrdersRequest;
import com.back.back9.domain.orders.dto.OrderResponse;
import com.back.back9.domain.orders.service.OrdersService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/orders")
@RequiredArgsConstructor
@Validated
@Slf4j
public class OrdersController {
    private final OrdersService ordersService;

    @GetMapping("/wallet/{wallet_id}")
    public ResponseEntity<List<OrderResponse>> getOrders(@PathVariable Long wallet_id) {
        return ResponseEntity.ok(ordersService.getOrdersByWalletId(wallet_id));
    }

    @PostMapping("/wallet/{wallet_id}")
    public ResponseEntity<OrderResponse> executeTrade(
            @PathVariable("wallet_id") Long walletId,
            @RequestBody OrdersRequest ordersRequest) {

        OrderResponse response = ordersService.executeTrade(walletId, ordersRequest);
        return ResponseEntity.ok(response);
    }
}

File: backend/src/main/kotlin/com/back/back9/domain/orders/controller/OrdersController.kt
package com.back.back9.domain.orders.controller

import com.back.back9.domain.orders.dto.OrderResponse
import com.back.back9.domain.orders.dto.OrdersRequest
import com.back.back9.domain.orders.service.OrdersService
import lombok.extern.slf4j.Slf4j
import org.springframework.http.ResponseEntity
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/orders")
@Validated
@Slf4j
class OrdersController(private val ordersService: OrdersService) {
    @GetMapping("/wallet/{wallet_id}")
    fun getOrders(@PathVariable wallet_id: Long?): ResponseEntity<List<OrderResponse?>?> {
        return ResponseEntity.ok<List<OrderResponse?>?>(ordersService.getOrdersByWalletId(wallet_id))
    }

    @PostMapping("/wallet/{wallet_id}")
    fun executeTrade(
        @PathVariable("wallet_id") walletId: Long,
        @RequestBody ordersRequest: OrdersRequest
    ): ResponseEntity<OrderResponse?> {
        val response = ordersService.executeTrade(walletId, ordersRequest)
        return ResponseEntity.ok<OrderResponse?>(response)
    }
}
